game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/FiringGroup.java

```diff
@@ -0,0 +1,143 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.Matches;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Value;
+import org.triplea.java.collections.CollectionUtils;
+
+/**
+ * Group of units that are firing on targets
+ *
+ * <p>If the group is suicideOnHit, then all of the units should have the same unit type
```

Comment:
This seems to imply a different object semantic, perhaps even a different object. WDYT of something like `SuicideOnHitFiringGroup`? Would it simplify the eventual usages?

Reply:
No, it wouldn't simplify any eventual usage.  This flag is only used to determine whether to kill the units after they fire.  We have to have a separate group so that we can easily detect which units hit and so which ones can be removed.